iothub/device/src/Transport/AmqpIot/AmqpIotCbsTokenProvider.cs

```diff
@@ -9,9 +9,10 @@
 
 namespace Microsoft.Azure.Devices.Client.Transport.AmqpIot
 {
-    internal class AmqpIotCbsTokenProvider : ICbsTokenProvider
+    internal class AmqpIotCbsTokenProvider : ICbsTokenProvider, IDisposable
```

Comment:
AmqpIotCbsTokenProvider contains an IDisposable type, so making it IDisposable as well